src/taskgraph/util/taskcluster.py

```diff
@@ -367,3 +368,22 @@ def list_task_group_incomplete_tasks(task_group_id):
             params = {"continuationToken": resp.get("continuationToken")}
         else:
             break
+
+
+def get_recursive_dependencies(
+    task_ids: List[str], add_given_tasks: bool = False, use_proxy: bool = False
```

Comment:
Is passing a list of ids here necessary? Could the caller just call this function in a loop instead?

Reply:
Oh I'm silly, it's to make the recursion easier. Makes sense.

Reply:
Actually might be nice to support this first argument as a simple string as well (which is probably the common case for the initial call).

Reply:
That's a good call, yeah.